packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.test.ts

```diff
@@ -0,0 +1,96 @@
+import { expect } from "chai";
+import { mergeI18nMessagesWithLocaleFileContent } from "./mergeI18nMessagesWithLocaleFile";
+
+describe("mergeI18nMessagesWithLocaleFileContent", () => {
+  it("do not add messages to locale without resources", () => {
+    const locale = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+    }`;
+    const localeExpected = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+      resources: {}
+    }`;
+    const localeResult = mergeI18nMessagesWithLocaleFileContent(locale, "en", {});
+    expect(localeResult).to.equal(localeExpected);
+  });
+
+  it("add messages to locale without resources", () => {
+    const locale = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+    }`;
+    const localeExpected = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+
+      resources: {
+        AnyDTO: {
+          newField: "newValue"
+        }
+      }
+    }`;
+    const localeResult = mergeI18nMessagesWithLocaleFileContent(locale, "en", {
+      AnyDTO: {
+        newField: "newValue"
+      }
+    });
+    expect(localeResult).to.equal(localeExpected);
+  });
+
+  it("add messages to locale with resources", () => {
+    const locale = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+      resources: {}
+    }`;
+    const localeExpected = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+      resources: {
+        AnyDTO: {
+          newField: "newValue"
+        }
+      }
+    }`;
+    const localeResult = mergeI18nMessagesWithLocaleFileContent(locale, "en", {
+      AnyDTO: {
+        newField: "newValue"
+      }
+    });
+    expect(localeResult).to.equal(localeExpected);
+  });
+
+ 
+  it("add new and default messages to locale with resources", () => {
+    const locale = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+      resources: {
+        AnyDTO: {
+          oldTouchedField: "oldValue",
+          oldUntouchedField: "oldValue",
+        }
+      }
+    }`;
+    const localeExpected = `
+    export const en: TranslationMessages = {
+      ...englishMessages,
+      resources: {
+        AnyDTO: {
+          newField: "newValue",
+          oldTouchedField: "oldValue",
+          oldUntouchedField: "oldValue"
+        }
+      }
+    }`;
+    const localeResult = mergeI18nMessagesWithLocaleFileContent(locale, "en", {
+      AnyDTO: {
+        newField: "newValue",
+        oldTouchedField: "newValue"
+      }
+    });
+    expect(localeResult).to.equal(localeExpected);
+  });
```

Comment:
If there is no 'en' object in file, error should be thrown
```suggestion
  it("should throw obvious error message if there is no 'en' object in file", function() {
    const locale = `
    export const incorrectLocalObjectName: TranslationMessages = {
      ...englishMessages,
    }`;
    // TODO Error should be thrown, error message should be clear
    mergeI18nMessagesWithLocaleFileContent(locale, "en", {});
  });
```